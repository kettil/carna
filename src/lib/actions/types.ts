import { Watermark, Watermarks } from 'istanbul-lib-report';
import { SpawnKillHandler } from '../utils/createSpawnKillHandler';

type BabelActionProps = {
  watch?: boolean;
  spawnKillHandler?: SpawnKillHandler;
};

type BabelNodeActionProps = {
  watch: boolean;
  script: string;
  watchPaths?: string[];
};

type CommitlintActionProps = {
  edit: string;
};

type CoverageWatermarkThreshold = Watermark | number | undefined;
type CoverageActionProps = {
  projects: string[];
  watermarks?: Partial<Record<keyof Watermarks, CoverageWatermarkThreshold>>;
};

type DepcheckActionProps = {
  path: string;
  ignorePackages?: string[];
};

type EslintActionProps = {
  write?: boolean;
  files?: string[];
};

type JestActionProps = {
  projects: string[];
  updateSnapshot?: boolean;
  runInBand?: boolean;
  verbose?: boolean;
  watch?: boolean;
};

type LicensecheckActionProps = {
  ignorePackages: string[];
  path?: string;
};

type PrettierActionProps = {
  write?: boolean;
  extension?: string;
  files?: string[];
};

type SemverActionProps = {
  path?: string;
};

type TscActionProps = {
  mode: 'type-check' | 'type-create';
};

export type {
  BabelActionProps,
  BabelNodeActionProps,
  CommitlintActionProps,
  CoverageActionProps,
  CoverageWatermarkThreshold,
  DepcheckActionProps,
  EslintActionProps,
  JestActionProps,
  LicensecheckActionProps,
  PrettierActionProps,
  SemverActionProps,
  TscActionProps,
};
