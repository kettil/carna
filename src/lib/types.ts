import { Logger } from './cli/logger';

export type NpmInstallMode = 'prod' | 'dev' | 'optional';

export type PropsGlobal = {
  readonly cwd: string;
  readonly tpl: string;
  readonly cfg: string;
  readonly log: Logger;
  readonly vvv: boolean;
};

export type Action<
  Props extends Record<string, unknown> | undefined = undefined,
  ActionReturn = void
> = Props extends undefined
  ? (argv: PropsGlobal) => Promise<ActionReturn>
  : (argv: PropsGlobal, props: Props) => Promise<ActionReturn>;
