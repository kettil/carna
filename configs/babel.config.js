module.exports = {
  ignore: ['**/*.test.js', '**/*.test.ts', '**/*.test.tsx', '**/*.test-d.ts'],
  presets: ['@babel/env', '@babel/preset-typescript'],
  plugins: [
    [
      '@babel/plugin-transform-runtime', // replace helper code with runtime imports (deduplication)
      { corejs: 3 }, // import corejs polyfills exactly where they are needed
    ],
  ],
  env: {
    test: {
      // extra configuration for process.env.NODE_ENV === 'test'
      presets: ['@babel/env'], // overwrite env-config from above with transpiled module syntax
    },
  },
};
