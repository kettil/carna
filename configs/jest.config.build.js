/* eslint-disable import/no-unresolved, node/no-unpublished-require, node/no-missing-require */
module.exports = Object.assign(require('../../../jest.config.js'), {
  coverageReporters: ['text'],
});
