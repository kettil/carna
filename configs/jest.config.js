const { accessSync, constants } = require('fs');
const { join } = require('path');

const getSetupPathOrUndefined = (path, type, extendsion) => {
  try {
    const file = join(path, `${type}.${extendsion}`);

    accessSync(file, constants.R_OK);

    return [file];
  } catch (error) {
    return undefined;
  }
};

const getSetupFilesAfterEnvironment = (folder, type) => {
  const path = join(process.cwd(), 'tests', folder);

  return getSetupPathOrUndefined(path, type, 'ts') ?? getSetupPathOrUndefined(path, type, 'js');
};

const commons = {
  bail: 10,

  testEnvironment: 'node',

  clearMocks: true,
  restoreMocks: true,
};

const createProject = (folder, color, config) => ({
  ...commons,

  displayName: { name: folder, color },
  roots: ['<rootDir>/src', `<rootDir>/tests/${folder}`],

  testMatch: ['**/*.test.{js,ts,tsx}'],
  testPathIgnorePatterns: ['/node_modules/', '/src/'],

  setupFilesAfterEnv: getSetupFilesAfterEnvironment(folder, 'setupTests'),

  ...config,
});

const config = {
  ...commons,
  projects: [],

  collectCoverageFrom: [
    '<rootDir>/src/**/*.{ts,tsx,js}',
    '!<rootDir>/src/**/types.ts',
    '!<rootDir>/src/index.ts',

    '<rootDir>/packages/*/src/**/*.{ts,tsx,js}',
    '!<rootDir>/packages/*/src/**/types.ts',
    '!<rootDir>/packages/*/src/index.ts',
  ],
};

module.exports = { config, createProject };
