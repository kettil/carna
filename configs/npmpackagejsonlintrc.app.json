{
  "extends": "@kettil/npm-package-json-lint-config/app"
}
