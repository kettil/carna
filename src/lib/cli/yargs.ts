import { CommandModule, Arguments, Argv } from 'yargs';
import ExecutableError from '../errors/executableError';
import { PropsGlobal } from '../types';

export type CommandModuleCommand = string;

export type CommandModuleDescribe = NonNullable<CommandModule<PropsGlobal>['describe']>;

export type CommandModuleBuilder<Props = Record<string, unknown>> = (
  yargs: Argv<PropsGlobal>,
) => Argv<PropsGlobal & Props>;

export type CommandModuleHandler<Props = Record<string, unknown>> = (args: Arguments<PropsGlobal & Props>) => void;

export const builderDefault = <Props>(cmd: string, fn: CommandModuleBuilder<Props>): CommandModuleBuilder<Props> => (
  yargs,
) => fn(yargs).usage(`Usage: $0 ${cmd} [options]`).help().version(false)
  .showHelpOnFail(false);

export const errorHandler = (argv: PropsGlobal, error: unknown): void => {
  if (error instanceof ExecutableError) {
    argv.log.log(error.stdout);

    /* eslint-disable-next-line node/no-process-exit, unicorn/no-process-exit */
    process.exit(1);
  }

  throw error;
};
