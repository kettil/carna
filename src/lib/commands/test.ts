import { createBuilder, createHandler } from '../cli/yargs';
import testTask, { TestProps } from '../tasks/testTask';

export const command = 'test';
export const desc = 'Run the jest tests';

const options = { group: `${command}-Options` } as const;
const boolOptions = { ...options, type: 'boolean', default: false } as const;

export const handler = createHandler<TestProps>(testTask);
export const builder = createBuilder<TestProps>(command, (yargs) =>
  yargs.options({
    coverage: {
      ...boolOptions,
      alias: 'c',
      describe: 'Runs the coverage projects together and builds the coverage from them',
    },
    project: {
      ...options,
      array: true,
      type: 'string',
      alias: 'p',
      describe: 'Run only the tests of the specified projects',
    },
    updateSnapshot: {
      ...boolOptions,
      alias: 'u',
      describe: 'Use this flag to re-record every snapshot that fails during this test run',
    },
    runInBand: { ...boolOptions, alias: 'i', describe: 'Run all tests serially in the current process' },
    watch: { ...boolOptions, alias: 'w', describe: 'Watch files for changes and rerun tests related to changed files' },
  }),
);
