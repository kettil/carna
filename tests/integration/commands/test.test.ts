import * as fs from 'fs';
import { join } from 'path';
import { Argv } from 'yargs';
import { handler, builder, command, desc } from '../../../src/lib/commands/test';
import { PropsGlobal } from '../../../src/lib/types';
import { cwd, getArgv, getReadFileWithHooksFiles } from '../../shared/configs';

describe('command test', () => {
  test('it should be complete the yargs command structure', () => {
    expect(typeof command).toBe('string');
    expect(typeof desc).toBe('string');
    expect(typeof builder).toBe('function');
    expect(typeof handler).toBe('function');
  });

  test('it should be create the expected builder', () => {
    const yargs = {
      options: jest.fn((...args: unknown[]) => {
        expect(args).toMatchSnapshot('options');

        return yargs;
      }),
      usage: jest.fn((...args: unknown[]) => {
        expect(args).toMatchSnapshot('usage');

        return yargs;
      }),

      help: jest.fn(() => yargs),
      version: jest.fn(() => yargs),
      showHelpOnFail: jest.fn(() => yargs),
    } as unknown as Argv<PropsGlobal>;

    builder(yargs);
  });

  test('it should work', async () => {
    const result = await handler(getArgv({}));

    expect(result).toBeUndefined();
  });

  test('it should work with hooks', async () => {
    const coverageForUnit = {
      [join(cwd, 'src', 'index.ts')]: {
        path: join(cwd, 'src', 'index.ts'),
        statementMap: { 0: { start: { line: 4, column: 41 }, end: { line: 15, column: 1 } } },
        fnMap: {},
        branchMap: {},
        s: { 0: 1 },
        f: {},
        b: {},
      },

      [join(cwd, 'src', 'lib', 'app.ts')]: {
        path: join(cwd, 'src', 'lib', 'app.ts'),
        statementMap: {
          0: { start: { line: 10, column: 12 }, end: { line: 10, column: 52 } },
        },
        fnMap: {
          0: {
            name: '(anonymous_0)',
            decl: { start: { line: 19, column: 12 }, end: { line: 19, column: 13 } },
            loc: { start: { line: 19, column: 53 }, end: { line: 42, column: 1 } },
            line: 19,
          },
        },
        branchMap: {},
        s: { 0: 1, 1: 1, 2: 1, 3: 1, 4: 1, 5: 1, 6: 1 },
        f: { 0: 1, 1: 1 },
        b: {},
      },
    };

    // eslint-disable-next-line unicorn/prevent-abbreviations
    const coverageForE2e = {
      [join(cwd, 'src', 'lib', 'app.ts')]: {
        path: join(cwd, 'src', 'lib', 'app.ts'),
        statementMap: {},
        fnMap: {},
        branchMap: {},
        s: {},
        f: {},
        b: {},
      },
    };

    (fs as any).setMockReadFileFiles(
      getReadFileWithHooksFiles({
        [join(cwd, 'coverage', '_e2e', 'coverage-final.json')]: JSON.stringify(coverageForE2e),
        [join(cwd, 'coverage', '_unit', 'coverage-final.json')]: JSON.stringify(coverageForUnit),
      }),
    );

    const result = await handler(getArgv({}));

    expect(result).toBeUndefined();
  });

  test('it should work with ci mode', async () => {
    const result = await handler(getArgv({ ci: true }));

    expect(result).toBeUndefined();
  });

  test('it should work with watch mode', async () => {
    const result = await handler(getArgv({ watch: true }));

    expect(result).toBeUndefined();
  });

  test('it should work with update-snapshot mode', async () => {
    const result = await handler(getArgv({ updateSnapshot: true }));

    expect(result).toBeUndefined();
  });

  test('it should work with run-in-band mode', async () => {
    const result = await handler(getArgv({ runInBand: true }));

    expect(result).toBeUndefined();
  });

  test('it should work with selected project', async () => {
    const result = await handler(getArgv({ projects: ['e2e'] }));

    expect(result).toBeUndefined();
  });
});
