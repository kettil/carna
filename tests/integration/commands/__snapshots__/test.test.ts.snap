// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`command test it should be create the expected builder: options 1`] = `
Array [
  Object {
    "project": Object {
      "alias": "p",
      "array": true,
      "describe": "Run only the tests of the specified projects",
      "group": "test-Options",
      "type": "string",
    },
    "runInBand": Object {
      "alias": "i",
      "default": false,
      "describe": "Run all tests serially in the current process",
      "group": "test-Options",
      "type": "boolean",
    },
    "updateSnapshot": Object {
      "alias": "u",
      "default": false,
      "describe": "Use this flag to re-record every snapshot that fails during this test run",
      "group": "test-Options",
      "type": "boolean",
    },
    "watch": Object {
      "alias": "w",
      "default": false,
      "describe": "Watch files for changes and rerun tests related to changed files",
      "group": "test-Options",
      "type": "boolean",
    },
  },
]
`;

exports[`command test it should be create the expected builder: usage 1`] = `
Array [
  "Usage: $0 test [options]",
]
`;

exports[`command test it should work with ci mode: child-process 1`] = `
Object {
  "args": Array [
    "--colors",
    "--config",
    "/path/to/project/jest.config.js",
    "--selectProjects",
    "unit",
    "--runInBand",
    "--ci",
    "--silent",
    "--json",
    "--outputFile",
    "/path/to/project/coverage/_unit/jest-final.json",
    "--coverage",
    "--coverageDirectory",
    "/path/to/project/coverage/_unit",
    "--coverageReporters",
    "json",
    "--coverageThreshold",
    "'{\\"global\\":{\\"statements\\":0,\\"branches\\":0,\\"functions\\":0,\\"lines\\":0}}'",
  ],
  "command": "./node_modules/.bin/jest",
  "cwd": "/path/to/project",
}
`;

exports[`command test it should work with ci mode: child-process 2`] = `
Object {
  "args": Array [
    "--colors",
    "--config",
    "/path/to/project/jest.config.js",
    "--selectProjects",
    "e2e",
    "--runInBand",
    "--ci",
    "--silent",
    "--json",
    "--outputFile",
    "/path/to/project/coverage/_e2e/jest-final.json",
    "--coverage",
    "--coverageDirectory",
    "/path/to/project/coverage/_e2e",
    "--coverageReporters",
    "json",
    "--coverageThreshold",
    "'{\\"global\\":{\\"statements\\":0,\\"branches\\":0,\\"functions\\":0,\\"lines\\":0}}'",
  ],
  "command": "./node_modules/.bin/jest",
  "cwd": "/path/to/project",
}
`;

exports[`command test it should work with ci mode: fs 1`] = `
Object {
  "data": "{\\"/path/to/project/src/index.ts\\":{\\"path\\":\\"/path/to/project/src/index.ts\\",\\"statementMap\\":{\\"0\\":{\\"start\\":{\\"line\\":4,\\"column\\":41},\\"end\\":{\\"line\\":15,\\"column\\":1}}},\\"fnMap\\":{},\\"branchMap\\":{},\\"s\\":{\\"0\\":1},\\"f\\":{},\\"b\\":{}},\\"/path/to/project/src/lib/app.ts\\":{\\"path\\":\\"/path/to/project/src/lib/app.ts\\",\\"statementMap\\":{\\"0\\":{\\"start\\":{\\"line\\":10,\\"column\\":12},\\"end\\":{\\"line\\":10,\\"column\\":52}}},\\"fnMap\\":{\\"0\\":{\\"name\\":\\"(anonymous_0)\\",\\"decl\\":{\\"start\\":{\\"line\\":19,\\"column\\":12},\\"end\\":{\\"line\\":19,\\"column\\":13}},\\"loc\\":{\\"start\\":{\\"line\\":19,\\"column\\":53},\\"end\\":{\\"line\\":42,\\"column\\":1}},\\"line\\":19}},\\"branchMap\\":{},\\"s\\":{\\"0\\":1,\\"1\\":1,\\"2\\":1,\\"3\\":1,\\"4\\":1,\\"5\\":1,\\"6\\":1},\\"f\\":{\\"0\\":1,\\"1\\":1},\\"b\\":{}}}
",
  "path": "/path/to/project/coverage/coverage-final.json",
}
`;

exports[`command test it should work with ci mode: istanbul-reports 1`] = `
Array [
  "lcovonly",
  Object {},
]
`;

exports[`command test it should work with ci mode: istanbul-reports 2`] = `
Array [
  "json-summary",
  Object {},
]
`;

exports[`command test it should work with ci mode: istanbul-reports 3`] = `
Array [
  "text-summary",
  Object {},
]
`;

exports[`command test it should work with hooks: child-process 1`] = `
Object {
  "args": Array [
    "run",
    "precarna:test",
  ],
  "command": "npm",
  "cwd": "/path/to/project",
}
`;

exports[`command test it should work with hooks: child-process 2`] = `
Object {
  "args": Array [
    "--colors",
    "--config",
    "/path/to/project/jest.config.js",
    "--selectProjects",
    "unit",
    "--maxWorkers",
    "50%",
    "--coverage",
    "--coverageDirectory",
    "/path/to/project/coverage/_unit",
    "--coverageReporters",
    "json",
    "--coverageThreshold",
    "'{\\"global\\":{\\"statements\\":0,\\"branches\\":0,\\"functions\\":0,\\"lines\\":0}}'",
  ],
  "command": "./node_modules/.bin/jest",
  "cwd": "/path/to/project",
}
`;

exports[`command test it should work with hooks: child-process 3`] = `
Object {
  "args": Array [
    "--colors",
    "--config",
    "/path/to/project/jest.config.js",
    "--selectProjects",
    "e2e",
    "--maxWorkers",
    "50%",
    "--coverage",
    "--coverageDirectory",
    "/path/to/project/coverage/_e2e",
    "--coverageReporters",
    "json",
    "--coverageThreshold",
    "'{\\"global\\":{\\"statements\\":0,\\"branches\\":0,\\"functions\\":0,\\"lines\\":0}}'",
  ],
  "command": "./node_modules/.bin/jest",
  "cwd": "/path/to/project",
}
`;

exports[`command test it should work with hooks: child-process 4`] = `
Object {
  "args": Array [
    "run",
    "postcarna:test",
  ],
  "command": "npm",
  "cwd": "/path/to/project",
}
`;

exports[`command test it should work with hooks: fs 1`] = `
Object {
  "data": "{\\"/path/to/project/src/index.ts\\":{\\"path\\":\\"/path/to/project/src/index.ts\\",\\"statementMap\\":{\\"0\\":{\\"start\\":{\\"line\\":4,\\"column\\":41},\\"end\\":{\\"line\\":15,\\"column\\":1}}},\\"fnMap\\":{},\\"branchMap\\":{},\\"s\\":{\\"0\\":1},\\"f\\":{},\\"b\\":{}},\\"/path/to/project/src/lib/app.ts\\":{\\"path\\":\\"/path/to/project/src/lib/app.ts\\",\\"statementMap\\":{\\"0\\":{\\"start\\":{\\"line\\":10,\\"column\\":12},\\"end\\":{\\"line\\":10,\\"column\\":52}}},\\"fnMap\\":{\\"0\\":{\\"name\\":\\"(anonymous_0)\\",\\"decl\\":{\\"start\\":{\\"line\\":19,\\"column\\":12},\\"end\\":{\\"line\\":19,\\"column\\":13}},\\"loc\\":{\\"start\\":{\\"line\\":19,\\"column\\":53},\\"end\\":{\\"line\\":42,\\"column\\":1}},\\"line\\":19}},\\"branchMap\\":{},\\"s\\":{\\"0\\":1,\\"1\\":1,\\"2\\":1,\\"3\\":1,\\"4\\":1,\\"5\\":1,\\"6\\":1},\\"f\\":{\\"0\\":1,\\"1\\":1},\\"b\\":{}}}
",
  "path": "/path/to/project/coverage/coverage-final.json",
}
`;

exports[`command test it should work with hooks: istanbul-reports 1`] = `
Array [
  "lcovonly",
  Object {},
]
`;

exports[`command test it should work with hooks: istanbul-reports 2`] = `
Array [
  "json-summary",
  Object {},
]
`;

exports[`command test it should work with hooks: istanbul-reports 3`] = `
Array [
  "text-summary",
  Object {},
]
`;

exports[`command test it should work with hooks: istanbul-reports 4`] = `
Array [
  "html",
  Object {},
]
`;

exports[`command test it should work with run-in-band mode: child-process 1`] = `
Object {
  "args": Array [
    "--colors",
    "--config",
    "/path/to/project/jest.config.js",
    "--selectProjects",
    "unit",
    "--runInBand",
    "--coverage",
    "--coverageDirectory",
    "/path/to/project/coverage/_unit",
    "--coverageReporters",
    "json",
    "--coverageThreshold",
    "'{\\"global\\":{\\"statements\\":0,\\"branches\\":0,\\"functions\\":0,\\"lines\\":0}}'",
  ],
  "command": "./node_modules/.bin/jest",
  "cwd": "/path/to/project",
}
`;

exports[`command test it should work with run-in-band mode: child-process 2`] = `
Object {
  "args": Array [
    "--colors",
    "--config",
    "/path/to/project/jest.config.js",
    "--selectProjects",
    "e2e",
    "--runInBand",
    "--coverage",
    "--coverageDirectory",
    "/path/to/project/coverage/_e2e",
    "--coverageReporters",
    "json",
    "--coverageThreshold",
    "'{\\"global\\":{\\"statements\\":0,\\"branches\\":0,\\"functions\\":0,\\"lines\\":0}}'",
  ],
  "command": "./node_modules/.bin/jest",
  "cwd": "/path/to/project",
}
`;

exports[`command test it should work with run-in-band mode: fs 1`] = `
Object {
  "data": "{\\"/path/to/project/src/index.ts\\":{\\"path\\":\\"/path/to/project/src/index.ts\\",\\"statementMap\\":{\\"0\\":{\\"start\\":{\\"line\\":4,\\"column\\":41},\\"end\\":{\\"line\\":15,\\"column\\":1}}},\\"fnMap\\":{},\\"branchMap\\":{},\\"s\\":{\\"0\\":1},\\"f\\":{},\\"b\\":{}},\\"/path/to/project/src/lib/app.ts\\":{\\"path\\":\\"/path/to/project/src/lib/app.ts\\",\\"statementMap\\":{\\"0\\":{\\"start\\":{\\"line\\":10,\\"column\\":12},\\"end\\":{\\"line\\":10,\\"column\\":52}}},\\"fnMap\\":{\\"0\\":{\\"name\\":\\"(anonymous_0)\\",\\"decl\\":{\\"start\\":{\\"line\\":19,\\"column\\":12},\\"end\\":{\\"line\\":19,\\"column\\":13}},\\"loc\\":{\\"start\\":{\\"line\\":19,\\"column\\":53},\\"end\\":{\\"line\\":42,\\"column\\":1}},\\"line\\":19}},\\"branchMap\\":{},\\"s\\":{\\"0\\":1,\\"1\\":1,\\"2\\":1,\\"3\\":1,\\"4\\":1,\\"5\\":1,\\"6\\":1},\\"f\\":{\\"0\\":1,\\"1\\":1},\\"b\\":{}}}
",
  "path": "/path/to/project/coverage/coverage-final.json",
}
`;

exports[`command test it should work with run-in-band mode: istanbul-reports 1`] = `
Array [
  "lcovonly",
  Object {},
]
`;

exports[`command test it should work with run-in-band mode: istanbul-reports 2`] = `
Array [
  "json-summary",
  Object {},
]
`;

exports[`command test it should work with run-in-band mode: istanbul-reports 3`] = `
Array [
  "text-summary",
  Object {},
]
`;

exports[`command test it should work with run-in-band mode: istanbul-reports 4`] = `
Array [
  "html",
  Object {},
]
`;

exports[`command test it should work with selected project: child-process 1`] = `
Object {
  "args": Array [
    "--colors",
    "--config",
    "/path/to/project/jest.config.js",
    "--selectProjects",
    "e2e",
    "--maxWorkers",
    "50%",
    "--coverage",
    "--coverageDirectory",
    "/path/to/project/coverage/_e2e",
    "--coverageReporters",
    "json",
    "--coverageThreshold",
    "'{\\"global\\":{\\"statements\\":0,\\"branches\\":0,\\"functions\\":0,\\"lines\\":0}}'",
  ],
  "command": "./node_modules/.bin/jest",
  "cwd": "/path/to/project",
}
`;

exports[`command test it should work with selected project: fs 1`] = `
Object {
  "data": "{\\"/path/to/project/src/lib/app.ts\\":{\\"path\\":\\"/path/to/project/src/lib/app.ts\\",\\"statementMap\\":{},\\"fnMap\\":{},\\"branchMap\\":{},\\"s\\":{},\\"f\\":{},\\"b\\":{}}}
",
  "path": "/path/to/project/coverage/coverage-final.json",
}
`;

exports[`command test it should work with selected project: istanbul-reports 1`] = `
Array [
  "lcovonly",
  Object {},
]
`;

exports[`command test it should work with selected project: istanbul-reports 2`] = `
Array [
  "json-summary",
  Object {},
]
`;

exports[`command test it should work with selected project: istanbul-reports 3`] = `
Array [
  "text-summary",
  Object {},
]
`;

exports[`command test it should work with selected project: istanbul-reports 4`] = `
Array [
  "html",
  Object {},
]
`;

exports[`command test it should work with update-snapshot mode: child-process 1`] = `
Object {
  "args": Array [
    "--colors",
    "--config",
    "/path/to/project/jest.config.js",
    "--selectProjects",
    "unit",
    "--updateSnapshot",
    "--maxWorkers",
    "50%",
    "--coverage",
    "--coverageDirectory",
    "/path/to/project/coverage/_unit",
    "--coverageReporters",
    "json",
    "--coverageThreshold",
    "'{\\"global\\":{\\"statements\\":0,\\"branches\\":0,\\"functions\\":0,\\"lines\\":0}}'",
  ],
  "command": "./node_modules/.bin/jest",
  "cwd": "/path/to/project",
}
`;

exports[`command test it should work with update-snapshot mode: child-process 2`] = `
Object {
  "args": Array [
    "--colors",
    "--config",
    "/path/to/project/jest.config.js",
    "--selectProjects",
    "e2e",
    "--updateSnapshot",
    "--maxWorkers",
    "50%",
    "--coverage",
    "--coverageDirectory",
    "/path/to/project/coverage/_e2e",
    "--coverageReporters",
    "json",
    "--coverageThreshold",
    "'{\\"global\\":{\\"statements\\":0,\\"branches\\":0,\\"functions\\":0,\\"lines\\":0}}'",
  ],
  "command": "./node_modules/.bin/jest",
  "cwd": "/path/to/project",
}
`;

exports[`command test it should work with update-snapshot mode: fs 1`] = `
Object {
  "data": "{\\"/path/to/project/src/index.ts\\":{\\"path\\":\\"/path/to/project/src/index.ts\\",\\"statementMap\\":{\\"0\\":{\\"start\\":{\\"line\\":4,\\"column\\":41},\\"end\\":{\\"line\\":15,\\"column\\":1}}},\\"fnMap\\":{},\\"branchMap\\":{},\\"s\\":{\\"0\\":1},\\"f\\":{},\\"b\\":{}},\\"/path/to/project/src/lib/app.ts\\":{\\"path\\":\\"/path/to/project/src/lib/app.ts\\",\\"statementMap\\":{\\"0\\":{\\"start\\":{\\"line\\":10,\\"column\\":12},\\"end\\":{\\"line\\":10,\\"column\\":52}}},\\"fnMap\\":{\\"0\\":{\\"name\\":\\"(anonymous_0)\\",\\"decl\\":{\\"start\\":{\\"line\\":19,\\"column\\":12},\\"end\\":{\\"line\\":19,\\"column\\":13}},\\"loc\\":{\\"start\\":{\\"line\\":19,\\"column\\":53},\\"end\\":{\\"line\\":42,\\"column\\":1}},\\"line\\":19}},\\"branchMap\\":{},\\"s\\":{\\"0\\":1,\\"1\\":1,\\"2\\":1,\\"3\\":1,\\"4\\":1,\\"5\\":1,\\"6\\":1},\\"f\\":{\\"0\\":1,\\"1\\":1},\\"b\\":{}}}
",
  "path": "/path/to/project/coverage/coverage-final.json",
}
`;

exports[`command test it should work with update-snapshot mode: istanbul-reports 1`] = `
Array [
  "lcovonly",
  Object {},
]
`;

exports[`command test it should work with update-snapshot mode: istanbul-reports 2`] = `
Array [
  "json-summary",
  Object {},
]
`;

exports[`command test it should work with update-snapshot mode: istanbul-reports 3`] = `
Array [
  "text-summary",
  Object {},
]
`;

exports[`command test it should work with update-snapshot mode: istanbul-reports 4`] = `
Array [
  "html",
  Object {},
]
`;

exports[`command test it should work with watch mode: child-process 1`] = `
Object {
  "args": Array [
    "--colors",
    "--config",
    "/path/to/project/jest.config.js",
    "--selectProjects",
    "unit",
    "e2e",
    "--watch",
    "--maxWorkers",
    "25%",
    "--coverage",
    "--coverageDirectory",
    "/path/to/project/coverage",
    "--coverageReporters",
    "json",
    "--coverageThreshold",
    "'{\\"global\\":{\\"statements\\":0,\\"branches\\":0,\\"functions\\":0,\\"lines\\":0}}'",
    "--coverageReporters",
    "json-summary",
    "--coverageReporters",
    "lcovonly",
    "--coverageReporters",
    "html",
  ],
  "command": "./node_modules/.bin/jest",
  "cwd": "/path/to/project",
}
`;

exports[`command test it should work: child-process 1`] = `
Object {
  "args": Array [
    "--colors",
    "--config",
    "/path/to/project/jest.config.js",
    "--selectProjects",
    "unit",
    "--maxWorkers",
    "50%",
    "--coverage",
    "--coverageDirectory",
    "/path/to/project/coverage/_unit",
    "--coverageReporters",
    "json",
    "--coverageThreshold",
    "'{\\"global\\":{\\"statements\\":0,\\"branches\\":0,\\"functions\\":0,\\"lines\\":0}}'",
  ],
  "command": "./node_modules/.bin/jest",
  "cwd": "/path/to/project",
}
`;

exports[`command test it should work: child-process 2`] = `
Object {
  "args": Array [
    "--colors",
    "--config",
    "/path/to/project/jest.config.js",
    "--selectProjects",
    "e2e",
    "--maxWorkers",
    "50%",
    "--coverage",
    "--coverageDirectory",
    "/path/to/project/coverage/_e2e",
    "--coverageReporters",
    "json",
    "--coverageThreshold",
    "'{\\"global\\":{\\"statements\\":0,\\"branches\\":0,\\"functions\\":0,\\"lines\\":0}}'",
  ],
  "command": "./node_modules/.bin/jest",
  "cwd": "/path/to/project",
}
`;

exports[`command test it should work: fs 1`] = `
Object {
  "data": "{\\"/path/to/project/src/index.ts\\":{\\"path\\":\\"/path/to/project/src/index.ts\\",\\"statementMap\\":{\\"0\\":{\\"start\\":{\\"line\\":4,\\"column\\":41},\\"end\\":{\\"line\\":15,\\"column\\":1}}},\\"fnMap\\":{},\\"branchMap\\":{},\\"s\\":{\\"0\\":1},\\"f\\":{},\\"b\\":{}},\\"/path/to/project/src/lib/app.ts\\":{\\"path\\":\\"/path/to/project/src/lib/app.ts\\",\\"statementMap\\":{\\"0\\":{\\"start\\":{\\"line\\":10,\\"column\\":12},\\"end\\":{\\"line\\":10,\\"column\\":52}}},\\"fnMap\\":{\\"0\\":{\\"name\\":\\"(anonymous_0)\\",\\"decl\\":{\\"start\\":{\\"line\\":19,\\"column\\":12},\\"end\\":{\\"line\\":19,\\"column\\":13}},\\"loc\\":{\\"start\\":{\\"line\\":19,\\"column\\":53},\\"end\\":{\\"line\\":42,\\"column\\":1}},\\"line\\":19}},\\"branchMap\\":{},\\"s\\":{\\"0\\":1,\\"1\\":1,\\"2\\":1,\\"3\\":1,\\"4\\":1,\\"5\\":1,\\"6\\":1},\\"f\\":{\\"0\\":1,\\"1\\":1},\\"b\\":{}}}
",
  "path": "/path/to/project/coverage/coverage-final.json",
}
`;

exports[`command test it should work: istanbul-reports 1`] = `
Array [
  "lcovonly",
  Object {},
]
`;

exports[`command test it should work: istanbul-reports 2`] = `
Array [
  "json-summary",
  Object {},
]
`;

exports[`command test it should work: istanbul-reports 3`] = `
Array [
  "text-summary",
  Object {},
]
`;

exports[`command test it should work: istanbul-reports 4`] = `
Array [
  "html",
  Object {},
]
`;
